chia/full_node/mempool_manager.py

```diff
@@ -314,26 +314,27 @@ async def add_spend_bundle(
             assert cost is not None
             return uint64(cost), MempoolInclusionStatus.SUCCESS, None
 
-        cost, status, err, item, remove_items = await self.validate_spend_bundle(new_spend, npc_result, spend_name)
-        if status == MempoolInclusionStatus.FAILED:
-            assert cost is None and item is None
-        elif status == MempoolInclusionStatus.SUCCESS:
-            assert cost is not None and item is not None
-            self.mempool.remove_from_pool(remove_items)
+        err, item, remove_items = await self.validate_spend_bundle(new_spend, npc_result, spend_name)
+        if err is None:
+            # No error, immediately add to mempool, after removing conflicting TXs.
+            assert item is not None
             self.mempool.add_to_pool(item)
-        elif status == MempoolInclusionStatus.PENDING:
-            assert cost is not None and item is not None
+            self.mempool.remove_from_pool(remove_items)
```

Comment:
the add and remove changed order here. I would think it's relevant, right?

Reply:
or maybe it doesn't matter anymore because of the separation of "add" and "validate"